### PR TITLE
Added OIDC MFA EndPoint

### PR DESCRIPTION
**Pull request by cocojoe**
**Public API** 
_Methods Added:_
* `login(withOTP otp: String, mfaToken: String)`

_Errors Added_
* `isMultifactorTokenInvalid`

**Tests**
Added Error Tests
Added Endpoint Tests

### CODE REVIEW

File: Auth0Tests/AuthenticationErrorSpec.swift [hunk 1]
```diff
@@ -275,6 +320,7 @@ class AuthenticationErrorSpecSharedExamplesConfiguration: QuickConfiguration {
             it("should not match any known error") {
                 expect(error.isMultifactorCodeInvalid).to(beFalse(), description: "should not match mfa invalid")
                 expect(error.isMultifactorEnrollRequired).to(beFalse(), description: "should not match mfa enroll")
+                expect(error.isMultifactorCodeInvalid).to(beFalse(), description: "should not match code invalid")
```
Comment (lbalmaceda) [inline Auth0Tests/AuthenticationErrorSpec.swift]: what about `isMultifactorTokenInvalid`?

File: Auth0Tests/AuthenticationSpec.swift [hunk 2]
```diff
@@ -141,6 +141,41 @@ class AuthenticationSpec: QuickSpec {
...
+            it("should fail login with bad otp") {
+                waitUntil(timeout: Timeout) { done in
+                    auth.login(withOTP: "bad_otp", mfaToken: MFAToken).start { result in
+                        expect(result).to(haveAuthenticationError(code: "invalid_grant", description: "Invalid otp_code."))
+                        done()
+                    }
+                }
+            }
+
+            it("should fail login with bad otp") {
```
Comment (lbalmaceda) [inline Auth0Tests/AuthenticationSpec.swift]: this one is a dupe of the above (i can't spot any diff)

Comment (cocojoe) [reply Auth0Tests/AuthenticationSpec.swift]: Have replaced with invalid mfa token test

File: Auth0Tests/AuthenticationErrorSpec.swift [hunk 3]
```diff
@@ -13,7 +13,7 @@
 // all copies or substantial portions of the Software.
 //
 // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,u
```
Comment (lbalmaceda) [inline Auth0Tests/AuthenticationErrorSpec.swift]: u what?

File: Auth0Tests/AuthenticationErrorSpec.swift [hunk 4]
```diff
@@ -178,6 +205,24 @@ class AuthenticationErrorSpec: QuickSpec {
...
+            it("should detect invalid mfa token") {
+                let values = [
+                    "error": "expired_token",
+                    "error_description": "mfa_token is expired"
+                ]
+                let error = AuthenticationError(info: values, statusCode: 401)
+                expect(error.isMultifactorTokenInvalid) == true
+            }
+
+            it("should detect invalid malformed mfa token") {
```
Comment (lbalmaceda) [inline Auth0Tests/AuthenticationErrorSpec.swift]: should detect invalid malformed mfa token -> should detect malformed mfa token

File: Auth0Tests/AuthenticationErrorSpec.swift (hunk 3 above)
Comment (cocojoe) [reply Auth0Tests/AuthenticationErrorSpec.swift]: is my keyboard, it has issues :(

**Pull request merged**